catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/QueryStatus.java

```diff
@@ -98,6 +103,18 @@ public boolean isDone() {
         return !(state == State.ACTIVE);
     }
 
+    public boolean hasReasons() {
+        return !reasons.isEmpty();
+    }
+
+    public List<String> getReasons() {
+        return Collections.unmodifiableList(reasons);
```

Comment:
With each query, a brand new set of `ProcessingDetails` are used. Once the system loads the reasons for failure, there is no valid cause for modification. Additional reasons can still be added using the provided `addReason` method.